pisek/opendata/lib.py

```diff
@@ -0,0 +1,252 @@
+                    ti.input_path(TEST_SEED).name
+                ):
+                    result[test_num].append(OpendataTestcaseInfo.from_testcase_info(ti))
+
+        return result
+
+    def get_testcase(
+        self,
+        name: str,
+        test: int,
+        seed: int | None,
+        input_path: str,
+        output_path: str,
+    ) -> "Testcase":
+        if not 0 <= test <= self._config.tests_count:
+            raise InvalidArgument(f"No test of number {test}")
+
+        testcase_infos = self._inputs_list()
+        if name not in testcase_infos:
+            raise InvalidArgument(f"There is no testcase of name '{name}'.")
+        testcase_info = testcase_infos[name]
+
+        if not self._config.test_sections[test].in_test(
+            testcase_info.input_path(TEST_SEED).name
+        ):
+            raise InvalidArgument(f"Testcase '{name}' is not in test {test}")
+
+        if seed == None and testcase_info.seeded:
+            raise InvalidArgument(f"No {seed} for seeded testcase.")
+        elif seed is not None and not testcase_info.seeded:
+            raise InvalidArgument(f"Seed for unseeded testcase.")
+
+        return Testcase(
+            self._path, self._config, testcase_info, test, seed, input_path, output_path
+        )
+
+
+class Testcase:
+    def __init__(
+        self,
+        task_path: str,
+        config: TaskConfig,
+        info: TestcaseInfo,
+        test: int,
+        seed: int | None,
+        input_path: str,
+        output_path: str,
+    ) -> None:
+        self._task_path = task_path
+        self._config = config
+        self._info = info
+        self._test = test
+        self._seed = seed
+        self._input_dst = input_path
+        self._output_dst = output_path
+
+        self._tmp_dir: str | None = None
+
+    @staticmethod
+    def _path_present(path: str | None) -> bool:
+        return path is not None and os.path.exists(path)
+
+    def _input_path(self) -> InputPath:
+        assert self._tmp_dir is not None
+        return InputPath(self._tmp_dir, self._info.input_path(self._seed).name)
+
+    def _output_path(self) -> OutputPath:
+        assert self._tmp_dir is not None
+        return OutputPath(
+            self._tmp_dir, self._info.input_path(self._seed).to_output().name
+        )
+
+    def _make_tmp_dir(self) -> None:
+        if self._tmp_dir is None:
+            self._tmp_dir = tempfile.mkdtemp(prefix="pisek_opendata_")
+
+    def _clear_tmp_dir(self) -> None:
+        assert self._tmp_dir is not None
+        shutil.rmtree(self._tmp_dir)
+        self._tmp_dir = None
+
+    def _run_pipeline(self, manager: JobManager) -> None:
+        run_pipeline(self._task_path, lambda _: DirectPipeline(manager), **ENV_ARGS)
+
+    def gen_input(self, clear: bool = True) -> None:
+        self._make_tmp_dir()
+
+        if self._path_present(self._input_dst):
+            return os.link(self._input_dst, self._input_path().path)
```

Comment:
This won't work unless the temporary directory and the provided input path is in the same filesystem.